src/main/webapp/app/exercises/quiz/manage/quiz-question-list-edit-existing.component.ts

```diff
@@ -170,16 +171,58 @@ export class QuizQuestionListEditExistingComponent implements OnChanges {
         if (!this.importFile) {
             return;
         }
+
+        const fileName = this.importFile.name;
+        const fileExtension = fileName.split('.').last()?.toLowerCase();
+
+        if (fileExtension === 'zip') {
+            await this.handleZipFile(this.importFile);
+        } else {
+            this.handleJsonFile(this.importFile);
+        }
+    }
+
+    handleJsonFile(file: File) {
         const fileReader = this.generateFileReader();
         fileReader.onload = () => this.onFileLoadImport(fileReader);
-        fileReader.readAsText(this.importFile);
+        fileReader.readAsText(file);
     }
 
-    async onFileLoadImport(fileReader: FileReader) {
+    async handleZipFile(file: File) {
```

Comment:
The file parameter for both functions is always `this.importFile` so it can be removed.